bundles/automation/org.eclipse.smarthome.automation.core/src/main/java/org/eclipse/smarthome/automation/core/RuleRegistryImpl.java

```diff
@@ -29,18 +29,13 @@
 public class RuleRegistryImpl extends AbstractRegistry<Rule, String>implements RuleRegistry {
 
     private RuleEngine ruleEngine;
-    private Set<String> disabledRuledSet;
+    private Set<String> disabledRuledSet = new HashSet(0);
     private Logger logger;
     private Storage<Boolean> disabledRulesStorage;
 
-    public RuleRegistryImpl(RuleEngine ruleManager, ManagedRuleProvider rp, Storage<Boolean> disabledRules) {
+    public RuleRegistryImpl(RuleEngine ruleManager) {
         logger = LoggerFactory.getLogger(getClass());
-        this.disabledRulesStorage = disabledRules;
-        disabledRuledSet = loadDisabledRuleMap();
         this.ruleEngine = ruleManager;
-        if (rp != null) {
-            addProvider(rp);
```

Comment:
The providers are added in the Activator, but there is a bug which will be fixed.